testsuite/tests/typing-layouts/allow_any.ml

```diff
@@ -0,0 +1,180 @@
+(* TEST
+ flags = "-extension layouts_beta";
+ expect;
+*)
+
+(* Baseline: if the jkind doesn't match, we should get an error. *)
+module Mismatched_no_attrs : sig
+  type t : float64
+end = struct
+  type t = string
+end
+[%%expect{|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type t = string
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t = string end
+       is not included in
+         sig type t : float64 end
+       Type declarations do not match:
+         type t = string
+       is not included in
+         type t : float64
+       The layout of the first is value
+         because it is the primitive type string.
+       But the layout of the first must be a sublayout of float64
+         because of the definition of t at line 2, characters 2-18.
+|}]
+
+(* On the other hand, if we set the correct attributes on both the impl and the intf, we
+   shouldn't get an error (though, obviously, this is completely unsound!) *)
+module Mismatched_with_both_attrs : sig
+  type t : float64
+  [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
+end = struct
+  type t = string
+  [@@unsafe_allow_any_kind_in_intf "I love segfaults"]
+end
+[%%expect{|
+module Mismatched_with_both_attrs : sig type t : float64 end
```

Comment:
I figured out how to do that (breadcrumbs: we have to add something to `Outcometree.out_type_decl` to remember what we did) but along the way convinced myself somewhat that we shouldn't; specifically, I'm not sure it makes the most sense to *tell* clients of a module "we have lied about the kind", especially when the whole idea is to use this feature to expose safe interfaces to unsafe stuff.

I was also talking with @tdelvecchio-jsc about this feature, and he felt that maybe we should just make this an attribute on the impl (with a kind annotation) and leave it off the intf entirely, and do this when checking kinds on type declarations rather than when checking kinds during module inclusion. I was somewhat convinced, for the same reason, but don't feel terribly strongly about it

Reply:
per discussion further down-thread, I'm going to resolve this.